rts/Lua/LuaUnsyncedCtrl.cpp

```diff
@@ -3382,6 +3384,226 @@ int LuaUnsyncedCtrl::SetBuildFacing(lua_State* L)
+	const char* caller,
+	const int idx
+) {
+	if (lua_istable(L, idx)) {
+		for (lua_pushnil(L); lua_next(L, idx) != 0; lua_pop(L, 1)) {
+			// "key" = value (table format of CommandNotify)
+			// ignore the "coded" key; not a boolean value
+			if (lua_israwstring(L, -2)) {
+				if (lua_isboolean(L, -1)) {
+					const bool value = lua_toboolean(L, -1);
+					switch (hashString(lua_tostring(L, -2))) {
+						case hashString("unbuildable"):
+							opts.unbuildable = value;
+							break;
+					}
+				} else if (lua_istable(L, -1)) {
+					float color[4];
+					LuaUtils::ParseFloatArray(L, -1, &color[0], 4);
+					switch (hashString(lua_tostring(L, -2))) {
+						case hashString("defaultColor"):
+							opts.defaultColor = &color[0];
+							break;
+						case hashString("buildableColor"):
+							opts.buildableColor = &color[0];
+							break;
+						case hashString("unbuildableColor"):
+							opts.unbuildableColor = &color[0];
+							break;
+						case hashString("featureColor"):
+							opts.featureColor = &color[0];
+							break;
+						case hashString("illegalColor"):
+							opts.illegalColor = &color[0];
+							break;
+					}
+				}
+			}
+		}
+	} else if (!lua_isnil(L, idx)) {
+		luaL_error(L, "%s(): bad options-argument type", caller);
+		return false;
+	}
+	return true;
+}
+
+
+/***
+ *
+ * @function Spring.DrawBuildSquare
+ * @number unitDefID
+ * @number x
+ * @number y
+ * @number z
+ * @number facing
+ * @bool[opt] unbuildable
+ * @tparam[opt] rgb buildableColor
+ * @tparam[opt] rgb unbuildableColor
+ * @tparam[opt] rgb featureColor
+ * @tparam[opt] rgb illegalColor
+ * @treturn nil
+ */
+int LuaUnsyncedCtrl::DrawBuildSquare(lua_State* L)
+{
+	const int args = lua_gettop(L); // number of arguments
+
+	BuildInfo buildInfo;
+	if (!ParseBuildInfo(L, buildInfo, __func__, 1)) {
+		return 0;
+	}
+
+	LuaBuildSquareOptions opts;
+	if (args >= 6) {
+		if (!ParseBuildSquareOptions(L, opts, __func__, 6)) {
+			return 0;
+		}
+	}
+
+	unitDrawer->AddLuaBuildSquare(buildInfo, opts);
+	return 0;
+}
+
+
+/*** Build Info spec
+ *
+ * @table buildInfoSpec
+ *
+ * Used when assigning multiple commands at once
+ *
+ * @number unitDefID
+ * @number x
+ * @number y
+ * @number z
+ * @number facing
+ */
+
+
+/***
+ *
+ * @function Spring.DrawBuildSquareArray
+ * @tparam {buildInfoSpec,...} buildInfoArray
+ * @bool[opt] unbuildable
+ * @tparam[opt] rgb buildableColor
+ * @tparam[opt] rgb unbuildableColor
+ * @tparam[opt] rgb featureColor
+ * @tparam[opt] rgb illegalColor
+ * @treturn nil
+ */
+int LuaUnsyncedCtrl::DrawBuildSquareArray(lua_State* L)
+{
+	const int args = lua_gettop(L); // number of arguments
+
+	std::vector<BuildInfo> buildInfos;
+	if (!ParseBuildInfoArray(L, buildInfos, __func__, 1)) {
+		return 0;
+	}
+
+	LuaBuildSquareOptions opts;
+	if (args >= 2) {
+		if (!ParseBuildSquareOptions(L, opts, __func__, 2)) {
+			return 0;
+		}
+	}
+
+	for (BuildInfo &buildInfo: buildInfos) {
+		unitDrawer->AddLuaBuildSquare(buildInfo, opts);
```

Comment:
If this is just a trivial loop then I'd get rid of the function, one can iterate in Lua.

Reply:
removed